### PR TITLE
[WIP] Job, task, taskflow log notifications

### PR DESCRIPTION
**Pull request by TristanWright**
(addresses #230)

### CODE REVIEW

File: girder/taskflow/server/models/taskflow.py [hunk 1]
```diff
@@ -61,6 +61,7 @@ def append_to_log(self, taskflow, log):
             }
         }
 
+        send_log_notification('taskflow', taskflow, log)
```
Comment (TristanWright) [inline girder/taskflow/server/models/taskflow.py]: I wasn't entirely sure if this is the right place to send the log notification from.

File: cumulus/common/girder.py [hunk 2]
```diff
@@ -51,37 +51,42 @@ def get_task_token(cluster=None):
...
         if user_access['level'] == AccessType.ADMIN:
-            create_status_notification(resource_name, notification, {
-                '_id': user_access['id']
-            })
+            create_notification(resource_name, notif_type, notification,
+                                {'_id': user_access['id']})
 
 
-def send_status_notification(resource_type, resource):
+def send_notification(resource_type, notif_type, resource):
```
Comment (cjh1) [inline cumulus/common/girder.py]: I would keep this signature and provide a implementation like you did for send_log_notification. That way you don't have to change every place the function is used.

File: girder/cumulus/server/models/job.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def create(self, user, job):
         job['userId'] = user['_id']
 
         self.save(job)
-        send_status_notification('job', job)
+        send_notification('job', 'status', job)
```
Comment (cjh1) [inline girder/cumulus/server/models/job.py]: As mentioned above, I would preserve the send_status_notification(...) signature

File: girder/taskflow/server/models/taskflow.py (hunk 1 above)
Comment (cjh1) [inline girder/taskflow/server/models/taskflow.py]: I would move it after the update, that way we wouldn't send out a notification if the update failed and the log wasn't really updated.

Comment (TristanWright) [inline girder/taskflow/server/models/taskflow.py]: move to `update_state` in the same file?

File: girder/taskflow/server/models/taskflow.py [hunk 4]
```diff
@@ -61,6 +61,7 @@ def append_to_log(self, taskflow, log):
...
         return self.update(query, update, multi=False)
```
Comment (cjh1) [inline girder/taskflow/server/models/taskflow.py]: No, after the call to update. So the end of the function would look something like this:

```
result = self.update(query, update, multi=False)
send_notification('taskflow', 'status', taskflow)

return result
```

Comment (TristanWright) [inline girder/taskflow/server/models/taskflow.py]: I see, I thought something more complex like if the db $push op failed, thanks.

**Comment by cjh1** [comment]
Fixing branch name ...

**Pull request closed without merging**